query-dsl/src/main/java/org/infinispan/query/dsl/QueryResult.java

```diff
@@ -17,14 +16,14 @@
 public interface QueryResult<E> {
 
    /**
-    * @return The number of hits from the query, ignoring pagination. When the query is non-indexed, for performance
-    * reasons, the hit count is not calculated and will return {@link OptionalLong#empty()}.
+    * @return The number of hits from the query, ignoring pagination.
     */
-   OptionalLong hitCount();   // todo [anistor] if cache size is an int, how can hit count be a wider type?
+   TotalHitCount hitCount();
```

Comment:
I wouldn't change the signature of this method. We should deprecate `hitCount()` and have `TotalHitCount hits()` instead

Reply:
Nice idea to deprecate it and introduce a new method for it.
Not sure `hits()` is a good name to return the `TotalHitCount`, since the hits are returned with the `QueryResult#list`. @tristantarrant  What do you think of `count` instead?